File: doc/examples/DocExamples/ProductGuide/Configuration.cs
using System;
using Stormpath.Configuration.Abstractions;
using Stormpath.SDK.Account;
using Stormpath.SDK.Cache;
using Stormpath.SDK.Client;

namespace examples.ProductGuide
{
    public class Configuration
    {
        public void DisableCaching()
        {
            #region code/csharp/configuration/disable_caching.cs
            var client = Clients.Builder()
                .SetCacheProvider(CacheProviders.Create().DisabledCache())
                .Build();
            #endregion
        }

        public void CustomCacheConfig()
        {
            #region code/csharp/configuration/custom_cache_config.cs
            var cacheProvider = CacheProviders.Create().InMemoryCache()
                .WithDefaultTimeToLive(TimeSpan.FromSeconds(120))
                .WithDefaultTimeToIdle(TimeSpan.FromSeconds(600))
                .WithCache(Caches.ForResource<IAccount>()
                    .WithTimeToLive(TimeSpan.FromSeconds(900))
                    .WithTimeToIdle(TimeSpan.FromSeconds(900)))
                .Build();

            var client = Clients.Builder()
                .SetCacheProvider(cacheProvider)
                .Build();
            #endregion
        }

        public void RedisCache()
        {
            #region code/csharp/configuration/redis_cache.cs
            var client = Clients.Builder()
                .SetCacheProvider(CacheProviders.Create().RedisCache()
                    .WithRedisConnection("localhost:6379")
                    .Build())
                .Build();
            #endregion
        }

        public void ApiCredentials()
        {
            #region code/csharp/configuration/api_credentials.cs
            var client = Clients.Builder()
                .SetApiKeyId("your_id_here")
                .SetApiKeySecret("your_secret_here")
                .Build();
            #endregion
        }

        public void ConfigureBaseUrl()
        {
            #region code/csharp/configuration/use_enterprise_url.cs
            var client = Clients.Builder()
                .SetBaseUrl("https://enterprise.stormpath.io/v1")
                .Build();
            #endregion
        }

        public void ConnectionTimeout()
        {
            #region code/csharp/configuration/connection_timeout.cs
            var client = Clients.Builder()
                .SetConnectionTimeout(60 * 1000) // in milliseconds
                .Build();
            #endregion
        }

        public void UseBasicAuth()
        {
            #region code/csharp/configuration/use_basic_auth.cs
            var client = Clients.Builder()
                .SetAuthenticationScheme(ClientAuthenticationScheme.Basic)
                .Build();
            #endregion
        }

        public void UseProxy()
        {
            #region code/csharp/configuration/use_proxy.cs
            var client = Clients.Builder()
                .SetProxy(new ClientProxyConfiguration()
                {
                    Host = "myproxy.example.com",
                    Port = 8088,
                    Username = "proxyuser",
                    Password = "proxypassword"
                })
                .Build();
            #endregion
        }
    }
}

File: doc/examples/DocExamplesVB/ProductGuide/Configuration.vb
Imports Stormpath.Configuration.Abstractions
Imports Stormpath.SDK.Account
Imports Stormpath.SDK.Cache
Imports Stormpath.SDK.Client

Public Class Configuration

    Public Sub DisableCaching()
#Region "code/vbnet/configuration/disable_caching.vb"
        Dim client = Clients.Builder() _
            .SetCacheProvider(CacheProviders.Create().DisabledCache()) _
            .Build()
#End Region
    End Sub

    Public Sub CustomCacheConfig()
#Region "code/vbnet/configuration/custom_cache_config.vb"
        Dim cacheProvider = CacheProviders.Create().InMemoryCache().
            WithDefaultTimeToLive(TimeSpan.FromSeconds(120)).
            WithDefaultTimeToIdle(TimeSpan.FromSeconds(600)).
            WithCache(Caches.ForResource(Of IAccount)() _
                .WithTimeToLive(TimeSpan.FromSeconds(900)) _
                .WithTimeToIdle(TimeSpan.FromSeconds(900))) _
            .Build()

        Dim client = Clients.Builder().SetCacheProvider(cacheProvider).Build()
#End Region
    End Sub

    Public Sub RedisCache()
#Region "code/vbnet/configuration/redis_cache.vb"
        Dim client = Clients.Builder() _
            .SetCacheProvider(CacheProviders.Create().RedisCache() _
                .WithRedisConnection("localhost:6379") _
                .Build()) _
            .Build()
#End Region
    End Sub

    Public Sub ApiCredentials()
#Region "code/vbnet/configuration/api_credentials.vb"
        Dim client = Clients.Builder() _
            .SetApiKeyId("your_id_here") _
            .SetApiKeySecret("your_secret_here") _
            .Build()
#End Region
    End Sub

    Public Sub ConfigureBaseUrl()
#Region "code/vbnet/configuration/use_enterprise_url.vb"
        Dim client = Clients.Builder() _
            .SetBaseUrl("https://enterprise.stormpath.io/v1") _
            .Build()
#End Region
    End Sub

    Public Sub ConnectionTimeout()
#Region "code/vbnet/configuration/connection_timeout.vb"
        ' Value is in milliseconds
        Dim client = Clients.Builder() _
            .SetConnectionTimeout(60 * 1000) _
            .Build()
#End Region
    End Sub

    Public Sub UseBasicAuth()
#Region "code/vbnet/configuration/use_basic_auth.vb"
        Dim client = Clients.Builder() _
            .SetAuthenticationScheme(ClientAuthenticationScheme.Basic) _
            .Build()
#End Region
    End Sub

    Public Sub UseProxy()
#Region "code/vbnet/configuration/use_proxy.vb"
        Dim client = Clients.Builder() _
            .SetProxy(New ClientProxyConfiguration() With
            {
                .Host = "myproxy.example.com",
                .Port = 8088,
                .Username = "proxyuser",
                .Password = "proxypassword"
            }).Build()
#End Region
    End Sub

End Class
